### ISSUE TITLE
Use stack's `ghc-pkg`

### ISSUE THREAD

**Issue by SmartHypercube** [IHaskell/IHaskell#1387]
If I use `ihaskell install --stack` to install kernel, should it uses stack's `ghc-pkg` instead of tries to find in `$PATH`?

**Comment by vaibhavsagar** [comment]
How do you propose using Stack's `ghc-pkg`?

**Comment by SmartHypercube** [comment]
`stack exec ghc-pkg` or set the search path for all tools to the output of `stack path --bin-path`?

**Comment by vaibhavsagar** [comment]
My understanding is that we are doing essentially this here: https://github.com/IHaskell/IHaskell/blob/465fded2f705cedbc791d210508ca9febc04a208/main/Main.hs#L143-L144

If you have ideas for making this more robust, let me know.

**Comment by SmartHypercube** [comment]
Strange... According to the code, kernel will try to run `stack`, and check whether `The Haskell Tool Stack` is printed. If so, it will run `stack exec env`. I checked that running `stack` does print the string, and running `stack exec env` will be very slow because Stack will download and install GHC (it hasn't been installed yet). However, when I create a notebook with IHaskell kernel, log immediately shows `ghc-pkg` cannot be found. Apparently `stack exec env` is not invoked.

Is there anything I can provide to help?

**Comment by vaibhavsagar** [comment]
Why does running `stack exec env` download and install GHC? That indicates that there is a version mismatch between the LTS in the `~/.stack/global-project.yaml` file and the LTS used to install IHaskell.

**Comment by SmartHypercube** [comment]
Sorry, my mistake.

After edited `~/.stack/global-project/stack.yaml` to match the LTS version (18.28), I re-tested these:
1. `stack exec env`, find the `PATH=...` line.
2. `export PATH=...`
3. Confirm that `ghc-pkg` is now in `PATH` and is functional.
4. Open a new shell where `PATH` has not been modified.
5. `jupyter-notebook`, connect, create an IHaskell notebook. It fails with:
```
[I 12:31:22.151 NotebookApp] Kernel started: 624c4434-4a6c-4ca0-bdde-0a387710d176, name: haskell
ihaskell: 
Ran commands: 
ghc-pkg check
which ghc-pkg

Exception: user error (shelly did not find ghc-pkg in the PATH: /home/app/.local/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games)
```

Seems IHaskell is not using the `PATH` from `stack exec env`?

**Comment by vaibhavsagar** [comment]
Did you install IHaskell with `ihaskell install --stack`?

**Comment by SmartHypercube** [comment]
Yes. This is the content of `~/.local/share/jupyter/kernels/haskell/kernel.json`:
```json
{"display_name":"Haskell","language":"haskell","argv":["/home/app/.local/bin/ihaskell","kernel","{connection_file}","--ghclib","/home/app/.stack/programs/x86_64-linux/ghc-tinfo6-8.10.7/lib/ghc-8.10.7","+RTS","-M3g","-N2","-RTS","--stack"]}
```

**Comment by SmartHypercube** [comment]
https://github.com/commercialhaskell/stack/commit/49bf07ce76e6c8e9e63a0a582e8a2ae0078e93b3
`stack` changed behavior in this commit.
Before: Print on stdout and exit with 0.
After: Print on stderr and exit with 1.

**Comment by SmartHypercube** [comment]
I made a PR for this. However, I also wish that when `--stack` is used, and `stack` detection failed, kernel would report an error and crash, instead of silently ignoring `--stack`. I'm not sure the best way to implement this.
